service/models.py

```diff
@@ -119,6 +120,35 @@ def find_or_404(cls, promotion_id):
         logger.info("Processing lookup or 404 for id %s ...", promotion_id)
         return cls.query.get_or_404(promotion_id)
 
+    @classmethod
+    def find_by_site_wide(cls, site_wide):
+        """ Find a Promotion by site wide """
+        logger.info(" Processing lookup for site_wide %s ...", site_wide)
+        return cls.query.filter(cls.is_site_wide == site_wide)
+
+    @classmethod
+    def find_by_query_string(cls, args):
+        """ Find a Promotion by query string """
+        logger.info(" Processing lookup based on query string %s ...", args)
+        filters = {}
+        for f in ['is_site_wide', 'promo_code', 'amount', 'promo_type']:
+            if f in args:
+                filters[f] = args.get(f)
+        data = cls.query.filter_by(**filters) if filters else cls.query
+        if 'start_date' in args:
+            data = data.filter(cls.start_date >= args.get('start_date'))
+        if 'end_date' in args:
+            data = data.filter(cls.end_date <= args.get('end_date'))
+        if 'duration' in args:
+            data = data.filter(cls.start_date + timedelta(days = int(args.get('duration'))) >= cls.end_date)
+        return data.all()
+
+    @classmethod
+    def find_by_start_date(cls, start_date):
```

Comment:
This isn't used by the API and we could get rid of it. The functionality is covered by `find_by_query_string` above.